llvm/lib/CodeGen/ReplaceWithVeclib.cpp

```diff
@@ -110,15 +110,17 @@ static bool replaceWithCallToVeclib(const TargetLibraryInfo &TLI,
 
   // OloadTys collects types used in scalar intrinsic overload name.
   SmallVector<Type *, 3> OloadTys;
-  if (!RetTy->isVoidTy() && isVectorIntrinsicWithOverloadTypeAtArg(IID, -1))
+  if (!RetTy->isVoidTy() &&
+      isVectorIntrinsicWithOverloadTypeAtArg(IID, -1, /*TTI=*/nullptr))
```

Comment:
Regarding adding a dependency on the `TargetTransformInfo` analysis pass to retrieve TTI. There is currently no code path that would execute and actually use `TTI` in the `VectorUtils.cpp` function. While I don't expect the pass is expensive to do, it is currently completely unnecessary work regardless.  So I think it is okay to defer until there is an actual use case. Which as mentioned, seems unlikely.